library/ssl_misc.h

```diff
@@ -985,11 +985,24 @@ struct mbedtls_ssl_handshake_params {
     } tls13_master_secrets;
 
     mbedtls_ssl_tls13_handshake_secrets tls13_hs_secrets;
+
 #if defined(MBEDTLS_SSL_EARLY_DATA)
     mbedtls_ssl_tls13_early_secrets tls13_early_secrets;
     /** TLS 1.3 transform for early data and handshake messages. */
     mbedtls_ssl_transform *transform_earlydata;
-#endif
+
+#if defined(MBEDTLS_SSL_SRV_C)
+    /** The state of server side early data  during handshake.
+     *  Available values are
+     *  - MBEDTLS_SSL_EARLY_DATA_SRV_STATE_NOT_RECEIVED
+     *  - MBEDTLS_SSL_EARLY_DATA_SRV_STATE_REJECTED
+     *  - MBEDTLS_SSL_EARLY_DATA_SRV_STATE_ACCEPTED
+     */
+    uint8_t early_data_state;
```

Comment:
This is identical (same semantic) to `early_data_status` on client side. Thus why not using the same context field (`early_data_status` in `mbedtls_ssl_context`) and the same macros MBEDTLS_SSL_EARLY_DATA_STATUS_*. We probably do not want to expose it through the `mbedtls_ssl_get_early_data_status` API on server side though.

Reply:
Forget that. I plan to share it but forget that after `ssl->early_data_status` got merged.